backend/src/bundles/user-details/user-details.repository.ts

```diff
@@ -31,6 +36,68 @@ class UserDetailsRepository implements Repository {
         throw new Error(ErrorMessages.NOT_IMPLEMENTED);
     }
 
+    public async searchUsers(
+        payload: UserDetailsSearchUsersRequestDto,
+    ): Promise<UserDetailsEntity[]> {
+        let query = this.userDetailsModel.query();
+
+        if (payload.search) {
+            query = query.where('fullName', 'ilike', `%${payload.search}%`);
+        }
+
+        if (payload.isHired) {
+            query = query.where('isHired', payload.isHired);
+        }
+
+        if (payload.jobTitle) {
+            query = query.where('jobTitle', payload.jobTitle);
+        }
+
+        if (payload.experienceYears) {
+            query = query.where('experienceYears', payload.experienceYears);
+        }
+
+        if (payload.hardSkills && payload.hardSkills.length > 0) {
+            query = query.whereExists(
+                this.userDetailsModel
+                    .relatedQuery('talentHardSkills')
+                    .whereIn(
+                        `talent_hard_skills.${TalentHardSkillsTableColumn.HARD_SKILL_ID} = ANY(?)`,
+                        [payload.hardSkills],
+                    ),
```

Comment:
Are you sure you correct use whereIn, when I write about it in previous request I want to not just change method, but rewrite query to whereIn method ?